tools/windows-node-installer/pkg/cloudprovider/azure/vm.go

```diff
@@ -0,0 +1,756 @@
+func getDiskClient(authorizer autorest.Authorizer, subscriptionID string) compute.DisksClient {
+	diskClient := compute.NewDisksClient(subscriptionID)
+	diskClient.Authorizer = authorizer
+	return diskClient
+}
+
+func check(err error, msg string) {
+	if err != nil {
+		log.Error(err, fmt.Sprintf(msg))
+	}
+}
+
+// verifyForResource verifies if the resource exists or not.
+func verifyForResource(resource string) bool {
+	if resource != "" {
+		return true
+	} else {
+		return false
+	}
+}
+
+// getvnetProfile gets the vnet Profile of the existing openshift cluster.
+func (az *AzureProvider) getvnetProfile(ctx context.Context) (vnetProfile network.VirtualNetwork) {
+	vnetList, err := az.vnetClient.List(ctx, az.resourceGroupName)
+	if err != nil {
+		fmt.Errorf("cannot get the vnetProfile info: %v", err)
+		return
+	}
+	vnetListValues := vnetList.Values()
+	vnetProfile = vnetListValues[0]
+	return vnetProfile
+}
+
+// extractResourceName captures the resource name omitting the other details.
+// document the function.
+func extractResourceName(rawresource string) (name string) {
+	resultList := strings.Split(rawresource, "/")
+	arrayLength := len(resultList)
+	name = resultList[arrayLength-1]
+	return
+}
+
+// createPublicIP creates the public IP for the instance
+func (az *AzureProvider) createPublicIP(ctx context.Context) (ip network.PublicIPAddress, err error) {
+	nodeLocation := *(az.getvnetProfile(ctx).Location)
+	future, err := az.ipClient.CreateOrUpdate(
+		ctx,
+		az.resourceGroupName,
+		az.IpName,
+		network.PublicIPAddress{
+			Name:     to.StringPtr(az.IpName),
+			Location: to.StringPtr(nodeLocation),
+			PublicIPAddressPropertiesFormat: &network.PublicIPAddressPropertiesFormat{
+				PublicIPAddressVersion:   network.IPv4,
+				PublicIPAllocationMethod: network.Static,
+			},
+		},
+	)
+
+	if err != nil {
+		return ip, fmt.Errorf("cannot create public ip address: %v", err)
+	}
+
+	err = future.WaitForCompletionRef(ctx, az.ipClient.Client)
+	if err != nil {
+		return ip, fmt.Errorf("cannot create public ip address: %v", err)
+	}
+
+	return future.Result(az.ipClient)
+}
+
+// createNIC creates the nic for the instance
+func (az *AzureProvider) createNIC(ctx context.Context, vnetName, subnetName, nsgName string) (nic network.Interface, err error) {
+	nodeLocation := *(az.getvnetProfile(ctx).Location)
+	subnet, err := az.subnetsClient.Get(ctx, az.resourceGroupName, vnetName, subnetName, "")
+	if err != nil {
+		fmt.Errorf("failed to get subnet: %v", err)
+	}
+
+	ip, err := az.ipClient.Get(ctx, az.resourceGroupName, az.IpName, "")
+
+	if err != nil {
+		fmt.Errorf("failed to get ip address: %v", err)
+	}
+
+	nicParams := network.Interface{
+		Name:     to.StringPtr(az.NicName),
+		Location: to.StringPtr(nodeLocation),
+		InterfacePropertiesFormat: &network.InterfacePropertiesFormat{
+			IPConfigurations: &[]network.InterfaceIPConfiguration{
+				{
+					Name: to.StringPtr("ipConfig1"),
```

Comment:
Why this hardcoding?

Reply:
Oops, I haven't checked this, will remove this hardcoding.